task_reactredux/task1/src/containers/CounterContainer.js

```diff
@@ -1,44 +1,89 @@
 import React from 'react';
 import Counter from '../views/Counter';
+import commandsEnum from '../Constants';
 
 class CounterContainer extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             count: 0
         }
-
-        this.increment = this.increment.bind(this);
-        this.decrement = this.decrement.bind(this);
-        this.reset = this.reset.bind(this);
     }
 
-    increment() {
+    increment = () => {
         this.setState({
             count: this.state.count + 1
         });
     }
 
-    decrement() {
+    decrement = () => {
         this.setState({
             count: this.state.count - 1
         });
     }
 
-    reset() {
+    reset = () => {
         this.setState({
             count: 0
         });
     }
 
+    checkNumberOnParity = (num) => {
+        return num % 2 === 0;
+    }
+
+    componentDidMount() {
+        console.log("Child: componentDidMount()");
+    }
+
+    componentDidUpdate() {
+        console.log("Child: componentDidUpdate()");
+    }
+
+    componentWillUnmount() {
+        console.log("Child: componentWillUnmount()");
+    }
+
+    UNSAFE_componentWillReceiveProps(prop) {
+        console.log("Child: componentWillReceiveProps()");
+
+        let counter = this.state.count;
+
+        if (prop.command === commandsEnum.add  && this.checkNumberOnParity(counter) && counter !== 0) {
+            this.setState({
+                count: counter + 1
+            });
+        } else if (prop.command === commandsEnum.del && !this.checkNumberOnParity(counter) && counter !== 0) {
```

Comment:
I think you could swap parity checking and equality to zero checking. If  `counter === 0` you could skip `checkNumberOnParity`